scripts/street_sign_recognizer.py

```diff
@@ -36,29 +44,125 @@ def process_image(self, msg):
         # draw bounding box rectangle
         cv2.rectangle(self.cv_image, left_top, right_bottom, color=(0, 0, 255), thickness=5)
 
+        # predict which sign it is
+        self.template_estimates = self.template_matcher.predict(cv2.cvtColor(cropped_sign, cv2.COLOR_BGR2GRAY))
+
     def sign_bounding_box(self):
         """
         Returns
         -------
         (left_top, right_bottom) where left_top and right_bottom are tuples of (x_pixel, y_pixel)
             defining topleft and bottomright corners of the bounding box
         """
-        # TODO: YOUR SOLUTION HERE
-        left_top = (200, 200)
-        right_bottom = (400, 400)
+        #Create list of points where the sign is
+        points = np.transpose(np.nonzero(self.binary_image))
+        #Get descriptors of boundry box
+        x,y,w,h = cv2.boundingRect(points)
+
+        #The processing underestimates sign area
```

Comment:
I like how your comments reflect some insights when developing the pipeline (i.e. like boundingRect underestimating sign area).  I like this higher level picture that motivates the code that follows.